examples/printerdemo_mcu/zephyr/src/slint-zephyr.cpp

```diff
@@ -61,6 +61,46 @@ struct k_unique_lock
     ~k_unique_lock() { k_mutex_unlock(mutex); }
     struct k_mutex *mutex = nullptr;
 };
+
+struct RotationInfo
+{
+    using RenderingRotation = slint::platform::SoftwareRenderer::RenderingRotation;
+    RenderingRotation rotation = RenderingRotation::NoRotation;
+    slint::PhysicalSize size;
+
+    bool is_transpose() const
+    {
+        return rotation == RenderingRotation::Rotate90 || rotation == RenderingRotation::Rotate270;
+    }
+
+    bool mirror_width() const
+    {
+        return rotation == RenderingRotation::Rotate180 || rotation == RenderingRotation::Rotate270;
+    }
+
+    bool mirror_height() const
+    {
+        return rotation == RenderingRotation::Rotate90 || rotation == RenderingRotation::Rotate180;
+    }
+};
```

Comment:
This and the `transformed` functions below are essentially copied from [software_renderer.rs](https://github.com/slint-ui/slint/blob/b1dab293665921d3c62e33058850f1f4b50e34c3/internal/core/software_renderer.rs#L85). I don't have any experience with exposing Rust code as C++, is this feasible to do?